expression/builtin_cast.go

```diff
@@ -227,6 +227,9 @@ func (c *castAsDecimalFunctionClass) getFunction(ctx sessionctx.Context, args []
 	} else {
 		argTp = args[0].GetType().EvalType()
 	}
+	if mysql.HasUnsignedFlag(args[0].GetType().Flag) {
```

Comment:
The result flag doesn't contain the `unsigned` flag even if the input is an unsigned integer:
```sql
drop table if exists t;
create table t(a bigint unsigned);
insert into t values(pow(2, 63));
```

In MySQL
```sql
MySQL(localhost:3306) > select cast(a as decimal(65,0)) from t;
Field   1:  `cast(a as decimal(65,0))`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       NEWDECIMAL
Collation:  binary (63)
Length:     66
Max_length: 19
Decimals:   0
Flags:      BINARY NUM


+--------------------------+
| cast(a as decimal(65,0)) |
+--------------------------+
|      9223372036854775808 |
+--------------------------+
1 row in set (0.00 sec)
```